clients/da-vinci-client/src/main/java/com/linkedin/davinci/store/rocksdb/RocksDBSstFileWriter.java

```diff
@@ -250,19 +271,40 @@ private void removeSSTFilesAfterCheckpointing() {
     }
   }
 
-  private void makeSureAllPreviousSSTFilesBeforeCheckpointingExist() {
+  private boolean doesAllPreviousSSTFilesBeforeCheckpointingExist() {
     if (lastFinishedSSTFileNo < 0) {
       LOGGER.info("Since last finished sst file no is negative, there is nothing to verify");
-      return;
+      return true;
     }
-    for (int cur = 0; cur <= lastFinishedSSTFileNo; ++cur) {
-      String sstFilePath = composeFullPathForSSTFile(cur);
+    int currFileNo = 0;
+    for (; currFileNo <= lastFinishedSSTFileNo; ++currFileNo) {
+      String sstFilePath = composeFullPathForSSTFile(currFileNo);
       File sstFile = new File(sstFilePath);
       if (!sstFile.exists()) {
-        throw new VeniceException(
-            "SST File: " + sstFilePath + " doesn't exist, but last finished sst file no is: " + lastFinishedSSTFileNo);
+        break;
       }
     }
+
+    if (currFileNo > lastFinishedSSTFileNo) {
+      LOGGER.info("Number of SST files matches with the checkpoint, continuing");
+      return true;
+    }
+
+    /**
+     * There is a mismatch in the number of created SST files vs the checkpointed
+     * number of SST files: This means that the ingestion started but the storage
+     * node crashed before OffsetRecord with EOP as true being synced.
+     *
+     * This can mean one of the below things:
+     * 1. Crash after ingestion completion: Number of files is 0 in that case
+     * 2. Crash during the ingestion: The current idea of the state of the system
```

Comment:
Nitpick: would recommend deleting "idea of the", which results in the same meaning but flows a bit better.